native/cocos/renderer/pipeline/Define.h

```diff
@@ -158,12 +159,16 @@ enum class CC_DLL RenderQueueSortMode {
 };
 CC_ENUM_CONVERSION_OPERATOR(RenderQueueSortMode)
 
-struct CC_DLL RenderQueueDesc {
+class CC_DLL RenderQueueDesc : public cc::RefCounted {
+public:
+    RenderQueueDesc() = default;
+    RenderQueueDesc(bool isTransparent, RenderQueueSortMode sortMode, const ccstd::vector<ccstd::string> &stages) // NOLINT
+    : isTransparent(isTransparent), sortMode(sortMode), stages(stages) {}
     bool isTransparent = false;
     RenderQueueSortMode sortMode = RenderQueueSortMode::FRONT_TO_BACK;
     ccstd::vector<ccstd::string> stages;
 };
-using RenderQueueDescList = ccstd::vector<RenderQueueDesc>;
+using RenderQueueDescList = ccstd::vector<IntrusivePtr<RenderQueueDesc>>;
```

Comment:
Why need to use IntrusivePtr? Can avoid it?

Reply:
This vector is shared with js, it may crash when GC if it's not a smart pointer